src/reconcile/rec_write.c

```diff
@@ -2682,7 +2681,7 @@ __rec_split_fixup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 	 * Fix up our caller's information.
 	 */
 	len = WT_PTRDIFF32(r->first_free, bnd->start);
-	if (len >= r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree))
+	if (len > r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree))
```

Comment:
I think the reason this check was written was to assert we didn't have a logic error in `__rec_split()`, where we switched from split-boundary accumulation to end-of-page accumulation incorrectly, which would result in a remnant that could have fit into a split-boundary. In other words, if a remnant ever fit into a split-boundary, there's a bug.

The paths into this code have changed since then, though, particularly with `SPLIT_MAX` being removed and subsequently re-added.

Anyway: the only way we have a remnant at all is if we have set `SPLIT_MAX`, that is, we've switched to accumulating to the end of the page. (If we haven't set `SPLIT_MAX`, to get here we'd have hit a split-boundary, had another split-chunk of room available in the page, but then still been unable to fit the next key/value item into the space available, falling through from the `SPLIT_BOUNDARY` case into the `SPLIT_MAX` case.

So, this test can only fire if we're accumulating to the end of the page and we returned out of the `SPLIT_BOUNDARY` case and accumulated a remnant, then hit the end of the space-available with an item that wouldn't fit.

That remnant must therefore be less than `split_size - WT_PAGE_HEADER_BYTE_SIZE(btree)` because we only switch to the end-of-page accumulation if the space-available is less than that amount, so the test of less-than-or-equal-to is correct, we should never see a remnant large enough to fill a split chunk.